--- git status ---
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   parkour_isaaclab/actuators/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_isaaclab/actuators/__pycache__/parkour_actuator_cfg.cpython-310.pyc
	modified:   parkour_isaaclab/actuators/__pycache__/parkour_actuator_pd.cpython-310.pyc
	modified:   parkour_isaaclab/envs/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_isaaclab/envs/__pycache__/parkour_manager_based_env.cpython-310.pyc
	modified:   parkour_isaaclab/envs/__pycache__/parkour_manager_based_env_cfg.cpython-310.pyc
	modified:   parkour_isaaclab/envs/__pycache__/parkour_manager_based_rl_env.cpython-310.pyc
	modified:   parkour_isaaclab/envs/__pycache__/parkour_manager_based_rl_env_cfg.cpython-310.pyc
	modified:   parkour_isaaclab/envs/__pycache__/parkour_ui.cpython-310.pyc
	modified:   parkour_isaaclab/envs/__pycache__/parkour_viewprot_camera_contoller.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/__pycache__/events.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/__pycache__/observations.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/__pycache__/rewards.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/__pycache__/terminations.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/actions_cfg.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/joint_actions.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/parkour_command_cfg.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/uniform_parkour_command.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/parkours/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/parkours/__pycache__/parkour_event.cpython-310.pyc
	modified:   parkour_isaaclab/envs/mdp/parkours/__pycache__/parkour_events_cfg.cpython-310.pyc
	modified:   parkour_isaaclab/managers/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_isaaclab/managers/__pycache__/parkour_manager.cpython-310.pyc
	modified:   parkour_isaaclab/managers/__pycache__/parkour_manager_term_cfg.cpython-310.pyc
	modified:   parkour_isaaclab/managers/__pycache__/parkour_reward_manager.cpython-310.pyc
	modified:   parkour_isaaclab/terrains/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_isaaclab/terrains/__pycache__/parkour_terrain_generator.cpython-310.pyc
	modified:   parkour_isaaclab/terrains/__pycache__/parkour_terrain_generator_cfg.cpython-310.pyc
	modified:   parkour_isaaclab/terrains/__pycache__/parkour_terrain_importer.cpython-310.pyc
	modified:   parkour_isaaclab/terrains/__pycache__/utils.cpython-310.pyc
	modified:   parkour_isaaclab/terrains/extreme_parkour/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_isaaclab/terrains/extreme_parkour/__pycache__/extreme_parkour_terrains_cfg.cpython-310.pyc
	modified:   parkour_isaaclab/terrains/extreme_parkour/__pycache__/extreme_parkour_terrians.cpython-310.pyc
	modified:   parkour_isaaclab/terrains/extreme_parkour/config/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_isaaclab/terrains/extreme_parkour/config/__pycache__/parkour.cpython-310.pyc
	modified:   parkour_tasks/parkour_tasks.egg-info/PKG-INFO
	modified:   parkour_tasks/parkour_tasks.egg-info/SOURCES.txt
	modified:   parkour_tasks/parkour_tasks/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_tasks/parkour_tasks/__pycache__/default_cfg.cpython-310.pyc
	modified:   parkour_tasks/parkour_tasks/extreme_parkour_task/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/parkour_mdp_cfg.cpython-310.pyc
	modified:   parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/parkour_teacher_cfg.cpython-310.pyc
	modified:   parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/__init__.cpython-310.pyc
	modified:   parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/parkour_rl_cfg.cpython-310.pyc
	modified:   parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/rsl_teacher_ppo_cfg.cpython-310.pyc
	modified:   parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/parkour_teacher_cfg.py
	modified:   scripts/rsl_rl/__pycache__/__init__.cpython-310.pyc
	modified:   scripts/rsl_rl/__pycache__/cli_args.cpython-310.pyc
	modified:   scripts/rsl_rl/__pycache__/exporter.cpython-310.pyc
	modified:   scripts/rsl_rl/__pycache__/vecenv_wrapper.cpython-310.pyc
	modified:   scripts/rsl_rl/modules/__pycache__/__init__.cpython-310.pyc
	modified:   scripts/rsl_rl/modules/__pycache__/actor_critic_with_encoder.cpython-310.pyc
	modified:   scripts/rsl_rl/modules/__pycache__/distillation_with_extractor.cpython-310.pyc
	modified:   scripts/rsl_rl/modules/__pycache__/on_policy_runner_with_extractor.cpython-310.pyc
	modified:   scripts/rsl_rl/modules/__pycache__/ppo_with_extractor.cpython-310.pyc
	modified:   scripts/rsl_rl/modules/feature_extractors/__pycache__/__init__.cpython-310.pyc
	modified:   scripts/rsl_rl/modules/feature_extractors/__pycache__/depth_backbone.cpython-310.pyc
	modified:   scripts/rsl_rl/modules/feature_extractors/__pycache__/estimator.cpython-310.pyc
	modified:   scripts/rsl_rl/modules/feature_extractors/__pycache__/state_encoder.cpython-310.pyc

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	Isaaclab_Parkour.egg-info/
	logs/
	outputs/
	parkour_tasks/parkour_tasks.egg-info/requires.txt

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/parkour_isaaclab/actuators/__pycache__/__init__.cpython-310.pyc b/parkour_isaaclab/actuators/__pycache__/__init__.cpython-310.pyc
index 35791c5..17c5ba9 100644
Binary files a/parkour_isaaclab/actuators/__pycache__/__init__.cpython-310.pyc and b/parkour_isaaclab/actuators/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_isaaclab/actuators/__pycache__/parkour_actuator_cfg.cpython-310.pyc b/parkour_isaaclab/actuators/__pycache__/parkour_actuator_cfg.cpython-310.pyc
index 42278bf..fee7847 100644
Binary files a/parkour_isaaclab/actuators/__pycache__/parkour_actuator_cfg.cpython-310.pyc and b/parkour_isaaclab/actuators/__pycache__/parkour_actuator_cfg.cpython-310.pyc differ
diff --git a/parkour_isaaclab/actuators/__pycache__/parkour_actuator_pd.cpython-310.pyc b/parkour_isaaclab/actuators/__pycache__/parkour_actuator_pd.cpython-310.pyc
index 3898f38..37672ef 100644
Binary files a/parkour_isaaclab/actuators/__pycache__/parkour_actuator_pd.cpython-310.pyc and b/parkour_isaaclab/actuators/__pycache__/parkour_actuator_pd.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/__pycache__/__init__.cpython-310.pyc b/parkour_isaaclab/envs/__pycache__/__init__.cpython-310.pyc
index ebd5074..7898f31 100644
Binary files a/parkour_isaaclab/envs/__pycache__/__init__.cpython-310.pyc and b/parkour_isaaclab/envs/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/__pycache__/parkour_manager_based_env.cpython-310.pyc b/parkour_isaaclab/envs/__pycache__/parkour_manager_based_env.cpython-310.pyc
index 58a4921..6c12a49 100644
Binary files a/parkour_isaaclab/envs/__pycache__/parkour_manager_based_env.cpython-310.pyc and b/parkour_isaaclab/envs/__pycache__/parkour_manager_based_env.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/__pycache__/parkour_manager_based_env_cfg.cpython-310.pyc b/parkour_isaaclab/envs/__pycache__/parkour_manager_based_env_cfg.cpython-310.pyc
index 10aa9c5..0a84916 100644
Binary files a/parkour_isaaclab/envs/__pycache__/parkour_manager_based_env_cfg.cpython-310.pyc and b/parkour_isaaclab/envs/__pycache__/parkour_manager_based_env_cfg.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/__pycache__/parkour_manager_based_rl_env.cpython-310.pyc b/parkour_isaaclab/envs/__pycache__/parkour_manager_based_rl_env.cpython-310.pyc
index 18c1f37..7f18dbc 100644
Binary files a/parkour_isaaclab/envs/__pycache__/parkour_manager_based_rl_env.cpython-310.pyc and b/parkour_isaaclab/envs/__pycache__/parkour_manager_based_rl_env.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/__pycache__/parkour_manager_based_rl_env_cfg.cpython-310.pyc b/parkour_isaaclab/envs/__pycache__/parkour_manager_based_rl_env_cfg.cpython-310.pyc
index 0de7e43..9b648a8 100644
Binary files a/parkour_isaaclab/envs/__pycache__/parkour_manager_based_rl_env_cfg.cpython-310.pyc and b/parkour_isaaclab/envs/__pycache__/parkour_manager_based_rl_env_cfg.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/__pycache__/parkour_ui.cpython-310.pyc b/parkour_isaaclab/envs/__pycache__/parkour_ui.cpython-310.pyc
index cc3960c..a2eab61 100644
Binary files a/parkour_isaaclab/envs/__pycache__/parkour_ui.cpython-310.pyc and b/parkour_isaaclab/envs/__pycache__/parkour_ui.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/__pycache__/parkour_viewprot_camera_contoller.cpython-310.pyc b/parkour_isaaclab/envs/__pycache__/parkour_viewprot_camera_contoller.cpython-310.pyc
index 2e880e4..0198d9f 100644
Binary files a/parkour_isaaclab/envs/__pycache__/parkour_viewprot_camera_contoller.cpython-310.pyc and b/parkour_isaaclab/envs/__pycache__/parkour_viewprot_camera_contoller.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/__pycache__/__init__.cpython-310.pyc b/parkour_isaaclab/envs/mdp/__pycache__/__init__.cpython-310.pyc
index db2854c..af70924 100644
Binary files a/parkour_isaaclab/envs/mdp/__pycache__/__init__.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/__pycache__/events.cpython-310.pyc b/parkour_isaaclab/envs/mdp/__pycache__/events.cpython-310.pyc
index 11eea17..eeedced 100644
Binary files a/parkour_isaaclab/envs/mdp/__pycache__/events.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/__pycache__/events.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/__pycache__/observations.cpython-310.pyc b/parkour_isaaclab/envs/mdp/__pycache__/observations.cpython-310.pyc
index add777a..bacb966 100644
Binary files a/parkour_isaaclab/envs/mdp/__pycache__/observations.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/__pycache__/observations.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/__pycache__/rewards.cpython-310.pyc b/parkour_isaaclab/envs/mdp/__pycache__/rewards.cpython-310.pyc
index 7c32b5c..a6fbda7 100644
Binary files a/parkour_isaaclab/envs/mdp/__pycache__/rewards.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/__pycache__/rewards.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/__pycache__/terminations.cpython-310.pyc b/parkour_isaaclab/envs/mdp/__pycache__/terminations.cpython-310.pyc
index 2110747..8829a07 100644
Binary files a/parkour_isaaclab/envs/mdp/__pycache__/terminations.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/__pycache__/terminations.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/__init__.cpython-310.pyc b/parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/__init__.cpython-310.pyc
index 3deac1b..c4fe1af 100644
Binary files a/parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/__init__.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/actions_cfg.cpython-310.pyc b/parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/actions_cfg.cpython-310.pyc
index 54aba5c..3f1e66c 100644
Binary files a/parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/actions_cfg.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/actions_cfg.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/joint_actions.cpython-310.pyc b/parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/joint_actions.cpython-310.pyc
index 5df0572..f38a1ab 100644
Binary files a/parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/joint_actions.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/parkour_actions/__pycache__/joint_actions.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/__init__.cpython-310.pyc b/parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/__init__.cpython-310.pyc
index 2058977..9ca6123 100644
Binary files a/parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/__init__.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/parkour_command_cfg.cpython-310.pyc b/parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/parkour_command_cfg.cpython-310.pyc
index ce87cc1..a75871f 100644
Binary files a/parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/parkour_command_cfg.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/parkour_command_cfg.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/uniform_parkour_command.cpython-310.pyc b/parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/uniform_parkour_command.cpython-310.pyc
index 5619457..bb2ea91 100644
Binary files a/parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/uniform_parkour_command.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/parkour_commands/__pycache__/uniform_parkour_command.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/parkours/__pycache__/__init__.cpython-310.pyc b/parkour_isaaclab/envs/mdp/parkours/__pycache__/__init__.cpython-310.pyc
index 1c69be6..542de00 100644
Binary files a/parkour_isaaclab/envs/mdp/parkours/__pycache__/__init__.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/parkours/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/parkours/__pycache__/parkour_event.cpython-310.pyc b/parkour_isaaclab/envs/mdp/parkours/__pycache__/parkour_event.cpython-310.pyc
index 0870b87..9ac8d4c 100644
Binary files a/parkour_isaaclab/envs/mdp/parkours/__pycache__/parkour_event.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/parkours/__pycache__/parkour_event.cpython-310.pyc differ
diff --git a/parkour_isaaclab/envs/mdp/parkours/__pycache__/parkour_events_cfg.cpython-310.pyc b/parkour_isaaclab/envs/mdp/parkours/__pycache__/parkour_events_cfg.cpython-310.pyc
index 647cd72..885d15c 100644
Binary files a/parkour_isaaclab/envs/mdp/parkours/__pycache__/parkour_events_cfg.cpython-310.pyc and b/parkour_isaaclab/envs/mdp/parkours/__pycache__/parkour_events_cfg.cpython-310.pyc differ
diff --git a/parkour_isaaclab/managers/__pycache__/__init__.cpython-310.pyc b/parkour_isaaclab/managers/__pycache__/__init__.cpython-310.pyc
index b961570..186502b 100644
Binary files a/parkour_isaaclab/managers/__pycache__/__init__.cpython-310.pyc and b/parkour_isaaclab/managers/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_isaaclab/managers/__pycache__/parkour_manager.cpython-310.pyc b/parkour_isaaclab/managers/__pycache__/parkour_manager.cpython-310.pyc
index fff635f..f142d71 100644
Binary files a/parkour_isaaclab/managers/__pycache__/parkour_manager.cpython-310.pyc and b/parkour_isaaclab/managers/__pycache__/parkour_manager.cpython-310.pyc differ
diff --git a/parkour_isaaclab/managers/__pycache__/parkour_manager_term_cfg.cpython-310.pyc b/parkour_isaaclab/managers/__pycache__/parkour_manager_term_cfg.cpython-310.pyc
index d8053d5..6a3f182 100644
Binary files a/parkour_isaaclab/managers/__pycache__/parkour_manager_term_cfg.cpython-310.pyc and b/parkour_isaaclab/managers/__pycache__/parkour_manager_term_cfg.cpython-310.pyc differ
diff --git a/parkour_isaaclab/managers/__pycache__/parkour_reward_manager.cpython-310.pyc b/parkour_isaaclab/managers/__pycache__/parkour_reward_manager.cpython-310.pyc
index 505655b..efd3c8d 100644
Binary files a/parkour_isaaclab/managers/__pycache__/parkour_reward_manager.cpython-310.pyc and b/parkour_isaaclab/managers/__pycache__/parkour_reward_manager.cpython-310.pyc differ
diff --git a/parkour_isaaclab/terrains/__pycache__/__init__.cpython-310.pyc b/parkour_isaaclab/terrains/__pycache__/__init__.cpython-310.pyc
index d38f6df..7ba472f 100644
Binary files a/parkour_isaaclab/terrains/__pycache__/__init__.cpython-310.pyc and b/parkour_isaaclab/terrains/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_isaaclab/terrains/__pycache__/parkour_terrain_generator.cpython-310.pyc b/parkour_isaaclab/terrains/__pycache__/parkour_terrain_generator.cpython-310.pyc
index 60beae6..af1ebac 100644
Binary files a/parkour_isaaclab/terrains/__pycache__/parkour_terrain_generator.cpython-310.pyc and b/parkour_isaaclab/terrains/__pycache__/parkour_terrain_generator.cpython-310.pyc differ
diff --git a/parkour_isaaclab/terrains/__pycache__/parkour_terrain_generator_cfg.cpython-310.pyc b/parkour_isaaclab/terrains/__pycache__/parkour_terrain_generator_cfg.cpython-310.pyc
index 1732196..b92c734 100644
Binary files a/parkour_isaaclab/terrains/__pycache__/parkour_terrain_generator_cfg.cpython-310.pyc and b/parkour_isaaclab/terrains/__pycache__/parkour_terrain_generator_cfg.cpython-310.pyc differ
diff --git a/parkour_isaaclab/terrains/__pycache__/parkour_terrain_importer.cpython-310.pyc b/parkour_isaaclab/terrains/__pycache__/parkour_terrain_importer.cpython-310.pyc
index f3fc89f..2948b75 100644
Binary files a/parkour_isaaclab/terrains/__pycache__/parkour_terrain_importer.cpython-310.pyc and b/parkour_isaaclab/terrains/__pycache__/parkour_terrain_importer.cpython-310.pyc differ
diff --git a/parkour_isaaclab/terrains/__pycache__/utils.cpython-310.pyc b/parkour_isaaclab/terrains/__pycache__/utils.cpython-310.pyc
index 6d6b74a..c691653 100644
Binary files a/parkour_isaaclab/terrains/__pycache__/utils.cpython-310.pyc and b/parkour_isaaclab/terrains/__pycache__/utils.cpython-310.pyc differ
diff --git a/parkour_isaaclab/terrains/extreme_parkour/__pycache__/__init__.cpython-310.pyc b/parkour_isaaclab/terrains/extreme_parkour/__pycache__/__init__.cpython-310.pyc
index b83a15e..b47274d 100644
Binary files a/parkour_isaaclab/terrains/extreme_parkour/__pycache__/__init__.cpython-310.pyc and b/parkour_isaaclab/terrains/extreme_parkour/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_isaaclab/terrains/extreme_parkour/__pycache__/extreme_parkour_terrains_cfg.cpython-310.pyc b/parkour_isaaclab/terrains/extreme_parkour/__pycache__/extreme_parkour_terrains_cfg.cpython-310.pyc
index c82f2a8..4931d84 100644
Binary files a/parkour_isaaclab/terrains/extreme_parkour/__pycache__/extreme_parkour_terrains_cfg.cpython-310.pyc and b/parkour_isaaclab/terrains/extreme_parkour/__pycache__/extreme_parkour_terrains_cfg.cpython-310.pyc differ
diff --git a/parkour_isaaclab/terrains/extreme_parkour/__pycache__/extreme_parkour_terrians.cpython-310.pyc b/parkour_isaaclab/terrains/extreme_parkour/__pycache__/extreme_parkour_terrians.cpython-310.pyc
index 6fd68b3..d7d4e23 100644
Binary files a/parkour_isaaclab/terrains/extreme_parkour/__pycache__/extreme_parkour_terrians.cpython-310.pyc and b/parkour_isaaclab/terrains/extreme_parkour/__pycache__/extreme_parkour_terrians.cpython-310.pyc differ
diff --git a/parkour_isaaclab/terrains/extreme_parkour/config/__pycache__/__init__.cpython-310.pyc b/parkour_isaaclab/terrains/extreme_parkour/config/__pycache__/__init__.cpython-310.pyc
index 3516132..110363e 100644
Binary files a/parkour_isaaclab/terrains/extreme_parkour/config/__pycache__/__init__.cpython-310.pyc and b/parkour_isaaclab/terrains/extreme_parkour/config/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_isaaclab/terrains/extreme_parkour/config/__pycache__/parkour.cpython-310.pyc b/parkour_isaaclab/terrains/extreme_parkour/config/__pycache__/parkour.cpython-310.pyc
index 8e04828..f6051e6 100644
Binary files a/parkour_isaaclab/terrains/extreme_parkour/config/__pycache__/parkour.cpython-310.pyc and b/parkour_isaaclab/terrains/extreme_parkour/config/__pycache__/parkour.cpython-310.pyc differ
diff --git a/parkour_tasks/parkour_tasks.egg-info/PKG-INFO b/parkour_tasks/parkour_tasks.egg-info/PKG-INFO
index a55582a..07cbb6c 100644
--- a/parkour_tasks/parkour_tasks.egg-info/PKG-INFO
+++ b/parkour_tasks/parkour_tasks.egg-info/PKG-INFO
@@ -10,10 +10,12 @@ Classifier: Natural Language :: English
 Classifier: Programming Language :: Python :: 3.10
 Classifier: Isaac Sim :: 4.5.0
 Requires-Python: >=3.10
+Requires-Dist: fast_simplification
 Dynamic: author
 Dynamic: classifier
 Dynamic: home-page
 Dynamic: keywords
 Dynamic: maintainer
+Dynamic: requires-dist
 Dynamic: requires-python
 Dynamic: summary
diff --git a/parkour_tasks/parkour_tasks.egg-info/SOURCES.txt b/parkour_tasks/parkour_tasks.egg-info/SOURCES.txt
index 4e068b7..d5181b2 100644
--- a/parkour_tasks/parkour_tasks.egg-info/SOURCES.txt
+++ b/parkour_tasks/parkour_tasks.egg-info/SOURCES.txt
@@ -5,4 +5,5 @@ parkour_tasks.egg-info/PKG-INFO
 parkour_tasks.egg-info/SOURCES.txt
 parkour_tasks.egg-info/dependency_links.txt
 parkour_tasks.egg-info/not-zip-safe
+parkour_tasks.egg-info/requires.txt
 parkour_tasks.egg-info/top_level.txt
\ No newline at end of file
diff --git a/parkour_tasks/parkour_tasks/__pycache__/__init__.cpython-310.pyc b/parkour_tasks/parkour_tasks/__pycache__/__init__.cpython-310.pyc
index 6678364..97a88e1 100644
Binary files a/parkour_tasks/parkour_tasks/__pycache__/__init__.cpython-310.pyc and b/parkour_tasks/parkour_tasks/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_tasks/parkour_tasks/__pycache__/default_cfg.cpython-310.pyc b/parkour_tasks/parkour_tasks/__pycache__/default_cfg.cpython-310.pyc
index 5c14729..9476414 100644
Binary files a/parkour_tasks/parkour_tasks/__pycache__/default_cfg.cpython-310.pyc and b/parkour_tasks/parkour_tasks/__pycache__/default_cfg.cpython-310.pyc differ
diff --git a/parkour_tasks/parkour_tasks/extreme_parkour_task/__pycache__/__init__.cpython-310.pyc b/parkour_tasks/parkour_tasks/extreme_parkour_task/__pycache__/__init__.cpython-310.pyc
index c06b1ac..4ca7748 100644
Binary files a/parkour_tasks/parkour_tasks/extreme_parkour_task/__pycache__/__init__.cpython-310.pyc and b/parkour_tasks/parkour_tasks/extreme_parkour_task/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/__init__.cpython-310.pyc b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/__init__.cpython-310.pyc
index 65642b7..386fd99 100644
Binary files a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/__init__.cpython-310.pyc and b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/parkour_mdp_cfg.cpython-310.pyc b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/parkour_mdp_cfg.cpython-310.pyc
index dca2219..87b75ee 100644
Binary files a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/parkour_mdp_cfg.cpython-310.pyc and b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/parkour_mdp_cfg.cpython-310.pyc differ
diff --git a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/parkour_teacher_cfg.cpython-310.pyc b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/parkour_teacher_cfg.cpython-310.pyc
index 7edbd96..c25c88d 100644
Binary files a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/parkour_teacher_cfg.cpython-310.pyc and b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/__pycache__/parkour_teacher_cfg.cpython-310.pyc differ
diff --git a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/__init__.cpython-310.pyc b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/__init__.cpython-310.pyc
index 6891ec5..6414621 100644
Binary files a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/__init__.cpython-310.pyc and b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/__init__.cpython-310.pyc differ
diff --git a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/parkour_rl_cfg.cpython-310.pyc b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/parkour_rl_cfg.cpython-310.pyc
index 590be96..40d9a2c 100644
Binary files a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/parkour_rl_cfg.cpython-310.pyc and b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/parkour_rl_cfg.cpython-310.pyc differ
diff --git a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/rsl_teacher_ppo_cfg.cpython-310.pyc b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/rsl_teacher_ppo_cfg.cpython-310.pyc
index 371ec55..28c3d6a 100644
Binary files a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/rsl_teacher_ppo_cfg.cpython-310.pyc and b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/agents/__pycache__/rsl_teacher_ppo_cfg.cpython-310.pyc differ
diff --git a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/parkour_teacher_cfg.py b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/parkour_teacher_cfg.py
index 2741a07..c871815 100644
--- a/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/parkour_teacher_cfg.py
+++ b/parkour_tasks/parkour_tasks/extreme_parkour_task/config/go2/parkour_teacher_cfg.py
@@ -31,7 +31,7 @@ class ParkourTeacherSceneCfg(ParkourDefaultSceneCfg):
         
 @configclass
 class UnitreeGo2TeacherParkourEnvCfg(ParkourManagerBasedRLEnvCfg):
-    scene: ParkourTeacherSceneCfg = ParkourTeacherSceneCfg(num_envs=6144, env_spacing=1.)
+    scene: ParkourTeacherSceneCfg = ParkourTeacherSceneCfg(num_envs=128, env_spacing=1.)
     # Basic settings
     observations: TeacherObservationsCfg = TeacherObservationsCfg()
     actions: ActionsCfg = ActionsCfg()
diff --git a/scripts/rsl_rl/__pycache__/__init__.cpython-310.pyc b/scripts/rsl_rl/__pycache__/__init__.cpython-310.pyc
index 6acb492..a292029 100644
Binary files a/scripts/rsl_rl/__pycache__/__init__.cpython-310.pyc and b/scripts/rsl_rl/__pycache__/__init__.cpython-310.pyc differ
diff --git a/scripts/rsl_rl/__pycache__/cli_args.cpython-310.pyc b/scripts/rsl_rl/__pycache__/cli_args.cpython-310.pyc
index 03b4f26..c9d4e03 100644
Binary files a/scripts/rsl_rl/__pycache__/cli_args.cpython-310.pyc and b/scripts/rsl_rl/__pycache__/cli_args.cpython-310.pyc differ
diff --git a/scripts/rsl_rl/__pycache__/exporter.cpython-310.pyc b/scripts/rsl_rl/__pycache__/exporter.cpython-310.pyc
index d1dc3e9..ccb05d4 100644
Binary files a/scripts/rsl_rl/__pycache__/exporter.cpython-310.pyc and b/scripts/rsl_rl/__pycache__/exporter.cpython-310.pyc differ
diff --git a/scripts/rsl_rl/__pycache__/vecenv_wrapper.cpython-310.pyc b/scripts/rsl_rl/__pycache__/vecenv_wrapper.cpython-310.pyc
index 7ec4a3e..1377e27 100644
Binary files a/scripts/rsl_rl/__pycache__/vecenv_wrapper.cpython-310.pyc and b/scripts/rsl_rl/__pycache__/vecenv_wrapper.cpython-310.pyc differ
diff --git a/scripts/rsl_rl/modules/__pycache__/__init__.cpython-310.pyc b/scripts/rsl_rl/modules/__pycache__/__init__.cpython-310.pyc
index dd03936..54f3438 100644
Binary files a/scripts/rsl_rl/modules/__pycache__/__init__.cpython-310.pyc and b/scripts/rsl_rl/modules/__pycache__/__init__.cpython-310.pyc differ
diff --git a/scripts/rsl_rl/modules/__pycache__/actor_critic_with_encoder.cpython-310.pyc b/scripts/rsl_rl/modules/__pycache__/actor_critic_with_encoder.cpython-310.pyc
index adeed19..591c2ca 100644
Binary files a/scripts/rsl_rl/modules/__pycache__/actor_critic_with_encoder.cpython-310.pyc and b/scripts/rsl_rl/modules/__pycache__/actor_critic_with_encoder.cpython-310.pyc differ
diff --git a/scripts/rsl_rl/modules/__pycache__/distillation_with_extractor.cpython-310.pyc b/scripts/rsl_rl/modules/__pycache__/distillation_with_extractor.cpython-310.pyc
index b09528f..217bf16 100644
Binary files a/scripts/rsl_rl/modules/__pycache__/distillation_with_extractor.cpython-310.pyc and b/scripts/rsl_rl/modules/__pycache__/distillation_with_extractor.cpython-310.pyc differ
diff --git a/scripts/rsl_rl/modules/__pycache__/on_policy_runner_with_extractor.cpython-310.pyc b/scripts/rsl_rl/modules/__pycache__/on_policy_runner_with_extractor.cpython-310.pyc
index 4503366..8a72222 100644
Binary files a/scripts/rsl_rl/modules/__pycache__/on_policy_runner_with_extractor.cpython-310.pyc and b/scripts/rsl_rl/modules/__pycache__/on_policy_runner_with_extractor.cpython-310.pyc differ
diff --git a/scripts/rsl_rl/modules/__pycache__/ppo_with_extractor.cpython-310.pyc b/scripts/rsl_rl/modules/__pycache__/ppo_with_extractor.cpython-310.pyc
index 6e8058b..bdb7bc6 100644
Binary files a/scripts/rsl_rl/modules/__pycache__/ppo_with_extractor.cpython-310.pyc and b/scripts/rsl_rl/modules/__pycache__/ppo_with_extractor.cpython-310.pyc differ
diff --git a/scripts/rsl_rl/modules/feature_extractors/__pycache__/__init__.cpython-310.pyc b/scripts/rsl_rl/modules/feature_extractors/__pycache__/__init__.cpython-310.pyc
index 081d346..81d093b 100644
Binary files a/scripts/rsl_rl/modules/feature_extractors/__pycache__/__init__.cpython-310.pyc and b/scripts/rsl_rl/modules/feature_extractors/__pycache__/__init__.cpython-310.pyc differ
diff --git a/scripts/rsl_rl/modules/feature_extractors/__pycache__/depth_backbone.cpython-310.pyc b/scripts/rsl_rl/modules/feature_extractors/__pycache__/depth_backbone.cpython-310.pyc
index 376783d..c717dfb 100644
Binary files a/scripts/rsl_rl/modules/feature_extractors/__pycache__/depth_backbone.cpython-310.pyc and b/scripts/rsl_rl/modules/feature_extractors/__pycache__/depth_backbone.cpython-310.pyc differ
diff --git a/scripts/rsl_rl/modules/feature_extractors/__pycache__/estimator.cpython-310.pyc b/scripts/rsl_rl/modules/feature_extractors/__pycache__/estimator.cpython-310.pyc
index 72b5b43..faaacc3 100644
Binary files a/scripts/rsl_rl/modules/feature_extractors/__pycache__/estimator.cpython-310.pyc and b/scripts/rsl_rl/modules/feature_extractors/__pycache__/estimator.cpython-310.pyc differ
diff --git a/scripts/rsl_rl/modules/feature_extractors/__pycache__/state_encoder.cpython-310.pyc b/scripts/rsl_rl/modules/feature_extractors/__pycache__/state_encoder.cpython-310.pyc
index aeb2c8b..ee58716 100644
Binary files a/scripts/rsl_rl/modules/feature_extractors/__pycache__/state_encoder.cpython-310.pyc and b/scripts/rsl_rl/modules/feature_extractors/__pycache__/state_encoder.cpython-310.pyc differ